### ISSUE TITLE
False positive for `Style/EmptyElse` when `AllowComments: true` with elsif-clause

### ISSUE THREAD

**Issue by ydah** [rubocop/rubocop#13085]
## Expected behavior

We expect that the following codes are not offenses:
```ruby
if condition
  statement
elsif condition2
  statement2
else
  # some comment
end
```

## Actual behavior

Offenses occur:
```ruby
if condition
  statement
elsif condition2
  statement2
else
^^^^ Redundant `else`-clause.
  # some comment
end
```

## Steps to reproduce the problem

The following settings must be included in .rubocop.yml:

```yaml
Style/EmptyElse:
  AllowComments: true
```

## RuboCop version

```
❯ bundle exec rubocop -V                                     
1.65.0 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.0) [arm64-darwin23]
  - rubocop-performance 1.21.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.0.3
```
